templates/shared_services/sonatype-nexus/terraform/vm.tf

```diff
@@ -0,0 +1,156 @@
+resource "azurerm_network_interface" "nexus" {
+  name                = "nic-nexus-${var.tre_id}"
+  location            = data.azurerm_resource_group.rg.location
+  resource_group_name = local.core_resource_group_name
+
+  ip_configuration {
+    name                          = "primary"
+    subnet_id                     = data.azurerm_subnet.shared.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_private_dns_zone" "nexus" {
+  name                = "nexus-${var.tre_id}.${data.azurerm_resource_group.rg.location}.cloudapp.azure.com"
+  resource_group_name = local.core_resource_group_name
+
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "nexus_core_vnet" {
+  name                  = "nexuslink-core"
+  resource_group_name   = local.core_resource_group_name
+  private_dns_zone_name = azurerm_private_dns_zone.nexus.name
+  virtual_network_id    = data.azurerm_virtual_network.core.id
+}
+
+resource "azurerm_private_dns_a_record" "nexus_vm" {
+  name                = "@"
+  zone_name           = azurerm_private_dns_zone.nexus.name
+  resource_group_name = local.core_resource_group_name
+  ttl                 = 300
+  records             = [azurerm_linux_virtual_machine.nexus.private_ip_address]
+}
+
+resource "random_password" "nexus_vm_password" {
+  length           = 16
+  lower            = true
+  min_lower        = 1
+  upper            = true
+  min_upper        = 1
+  number           = true
+  min_numeric      = 1
+  special          = true
+  min_special      = 1
+  override_special = "_%@"
+}
+
+resource "random_password" "nexus_admin_password" {
+  length           = 16
+  lower            = true
+  min_lower        = 1
+  upper            = true
+  min_upper        = 1
+  number           = true
+  min_numeric      = 1
+  special          = true
+  min_special      = 1
+  override_special = "_%@"
+}
+
+resource "azurerm_key_vault_secret" "nexus_vm_password" {
+  name         = "nexus-vm-password"
+  value        = random_password.nexus_vm_password.result
+  key_vault_id = data.azurerm_key_vault.kv.id
+}
+
+resource "azurerm_key_vault_secret" "nexus_admin_password" {
+  name         = "nexus-admin-password"
+  value        = random_password.nexus_admin_password.result
+  key_vault_id = data.azurerm_key_vault.kv.id
+}
+
+resource "azurerm_user_assigned_identity" "nexus_msi" {
+  name                = "id-nexus-${var.tre_id}"
+  location            = data.azurerm_resource_group.rg.location
+  resource_group_name = local.core_resource_group_name
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_role_assignment" "kv_reader" {
+  scope                = data.azurerm_key_vault.kv.id
+  role_definition_name = "Key Vault Reader"
+  principal_id         = azurerm_user_assigned_identity.nexus_msi.principal_id
+}
```

Comment:
It was needed to give the msi a scope within the subscription otherwise the az cli login would fail, but now realised we can get around the with the `--allow-no-subscriptions` flag so have removed